tests/e2e/e2e_upgrade_gov_and_authz_test.go

```diff
@@ -0,0 +1,214 @@
+
+	"github.com/kava-labs/kava/tests/e2e/testutil"
+	"github.com/kava-labs/kava/tests/util"
+)
+
+const (
+	govModuleAcc       = "kava10d07y265gmmuvt4z0w9aw880jnsr700jxh8cq5"
+	communityModuleAcc = "kava17d2wax0zhjrrecvaszuyxdf5wcu5a0p4qlx3t5"
+	kavadistModuleAcc  = "kava1cj7njkw2g9fqx4e768zc75dp9sks8u9znxrf0w"
+)
+
+func (suite *IntegrationTestSuite) TestGovParamChanges() {
+	suite.SkipIfUpgradeDisabled()
+
+	beforeUpgradeCtx := util.CtxAtHeight(suite.UpgradeHeight - 1)
+	afterUpgradeCtx := util.CtxAtHeight(suite.UpgradeHeight)
+
+	// fetch gov parameters before upgrade
+	govBeforeParams, err := suite.Kava.Gov.Params(beforeUpgradeCtx, &govv1.QueryParamsRequest{ParamsType: "tallying"})
+	suite.Require().NoError(err)
+
+	// assert expected gov quorum before upgrade
+	suite.NotEqual(govBeforeParams.TallyParams.Quorum, "0.200000000000000000")
+
+	govAfterParams, err := suite.Kava.Gov.Params(afterUpgradeCtx, &govv1.QueryParamsRequest{ParamsType: "tallying"})
+	suite.Require().NoError(err)
+
+	// assert expected gov quorum after upgrade
+	suite.Equal(govAfterParams.TallyParams.Quorum, "0.200000000000000000")
+
+}
+
+func (suite *IntegrationTestSuite) TestAuthzParamChanges() {
+	suite.SkipIfUpgradeDisabled()
+
+	beforeUpgradeCtx := util.CtxAtHeight(suite.UpgradeHeight - 1)
+	afterUpgradeCtx := util.CtxAtHeight(suite.UpgradeHeight)
+
+	// fetch authz grants before upgrade
+	authzBeforeGrants, err := suite.Kava.Authz.Grants(beforeUpgradeCtx, &authz.QueryGrantsRequest{Granter: kavadistModuleAcc, Grantee: govModuleAcc, Pagination: &query.PageRequest{Limit: 1000, CountTotal: true}})
+	suite.Require().NoError(err)
+	suite.Require().Equal(authzBeforeGrants.Pagination.Total, uint64(len(authzBeforeGrants.Grants)), "expected all grants to have been requested")
+
+	// no kavadist -> gov grants
+	suite.Equal(0, len(authzBeforeGrants.Grants))
+
+	// fetch authz grants after upgrade
+	authzAfterGrants, err := suite.Kava.Authz.Grants(afterUpgradeCtx, &authz.QueryGrantsRequest{Granter: kavadistModuleAcc, Grantee: govModuleAcc, Pagination: &query.PageRequest{Limit: 1000, CountTotal: true}})
+	suite.Require().NoError(err)
+	suite.Require().Equal(authzAfterGrants.Pagination.Total, uint64(len(authzAfterGrants.Grants)), "expected all grants to have been requested")
+
+	// one kavadist -> gov grants
+	suite.Require().Equal(1, len(authzAfterGrants.Grants))
+
+	grant := authzAfterGrants.Grants[0]
+
+	var authorization authz.Authorization
+	suite.Kava.EncodingConfig.Marshaler.UnpackAny(grant.Authorization, &authorization)
+
+	genericAuthorization, ok := authorization.(*authz.GenericAuthorization)
+	suite.Require().True(ok, "expected generic authorization")
+
+	// kavadist allows gov to MsgSend it's funds
+	suite.Equal(sdk.MsgTypeURL(&banktypes.MsgSend{}), genericAuthorization.Msg)
+	// no expiration
+	var expectedExpiration *time.Time
+	suite.Equal(expectedExpiration, grant.Expiration)
+}
+
+func (suite *IntegrationTestSuite) TestModuleAccountGovTransfers() {
```

Comment:
🔥 

great test. though it won't pass on mainnet. we might want a way for the test suite to know to skip if running against a place it doesn't have power (or time) to pass a gov proposal.

problem for our future selves, perhaps.

Reply:
good catch, updated to only run on kvtool